modules/administration-guide/pages/repairing-the-devworkspace-operator-on-openshift.adoc

```diff
@@ -0,0 +1,80 @@
+:_content-type: PROCEDURE
+:navtitle: Repairing the {devworkspace} Operator on OpenShift
+:description: Repairing the {devworkspace} Operator on OpenShift
+:keywords: devworkspace, operator, repair
+:page-aliases:
+
+[id="repairing-the-devworkspace-operator-on-openshift_{context}"]
+= Repairing the {devworkspace} Operator on OpenShift
+
+Under certain conditions, such as https://docs.openshift.com/container-platform/4.10/operators/understanding/olm/olm-understanding-olm.html[OLM] restart or cluster upgrade, the {prod2} Operator for {prod-short} might automatically install the {devworkspace} Operator even when it is already present on the cluster. In that case, you can repair the {devworkspace} Operator on OpenShift as follows:
+
+.Prerequisites
+
+* An active `{orch-cli}` session as a cluster administrator to the destination OpenShift cluster. See {orch-cli-link}.
+
+* On the *Installed Operators* page of the {platforms-name} web console, you see multiple entries for the {devworkspace} Operator or one entry that is stuck in a loop of *Installing* and *InstallSucceeded*.
+
+.Procedure
+
+. Delete the `devworkspace-controller` namespace that contains the failing pod.
+
+. Update `DevWorkspace` and `DevWorkspaceTemplate` Custom Resource Definitions (CRD) by setting the conversion strategy to `None` and removing the next several lines of the removed strategy:
++
+[source,yaml]
+----
+spec:
+  ...
+  conversion:
+    strategy: None
+status:
+...
+----
++
+TIP: You can find and edit the `DevWorkspace` and `DevWorkspaceTemplate` CRDs in the *Administrator* perspective of the OpenShift web console by searching for `DevWorkspace` in menu:Administration[CustomResourceDefinitions].
++
+NOTE: The `DevWorkspaceOperatorConfig` and `DevWorkspaceRouting` CRDs  have the conversion strategy set to `None` by default.
+
+. Remove the {devworkspace} Operator subscription:
++
+----
+$ oc delete sub devworkspace-operator
+----
+
+. Get the {devworkspace} Operator CSVs in the __<devworkspace-operator.vX.Y.Z>__ format:
++
+----
+$ oc get csv | grep devworkspace
+----
+
+. Remove each {devworkspace} Operator CSV:
++
+[subs="+quotes"]
+----
+$ oc delete csv __<devworkspace-operator.vX.Y.Z>__ <1>
```

Comment:
I wasn't able to remove csv's:
```
$ oc get csv | grep devworkspace
devworkspace-operator.v0.15.2             DevWorkspace Operator          0.15.2                 devworkspace-operator.v0.15.1   Replacing
devworkspace-operator.v0.15.3             DevWorkspace Operator          0.15.3                 devworkspace-operator.v0.15.2   Pending
[ndp@fedora crw-jenkins]$ oc delete csv devworkspace-operator.v0.15.2
clusterserviceversion.operators.coreos.com "devworkspace-operator.v0.15.2" deleted
[ndp@fedora crw-jenkins]$ oc get csv | grep devworkspace
devworkspace-operator.v0.15.2             DevWorkspace Operator          0.15.2                 devworkspace-operator.v0.15.1   Replacing
devworkspace-operator.v0.15.3             DevWorkspace Operator          0.15.3                 devworkspace-operator.v0.15.2   Pending
[ndp@fedora crw-jenkins]$ oc delete csv devworkspace-operator.v0.15.3
clusterserviceversion.operators.coreos.com "devworkspace-operator.v0.15.3" deleted
[ndp@fedora crw-jenkins]$ oc get csv | grep devworkspace
devworkspace-operator.v0.15.2             DevWorkspace Operator          0.15.2                 devworkspace-operator.v0.15.1   Replacing
devworkspace-operator.v0.15.3             DevWorkspace Operator          0.15.3                 devworkspace-operator.v0.15.2   Pending
```

Reply:
it is not necessarily openshift-operators, but the namespace where DWO operator is installed.

Reply:
Fixing it as follows:
```
----
$ oc delete csv __<devworkspace-operator.vX.Y.Z>__ \ <1>
-n openshift-operators <2>
----
<1> For example, `devworkspace-operator.v0.15.1` or `devworkspace-operator.v0.15.2`.
<2> `openshift-operators` or an OpenShift {orch-namespace} where the {devworkspace} Operator is installed.
```

Reply:
And I've removed the first callout:
`<1> For example, `devworkspace-operator.v0.15.1` or `devworkspace-operator.v0.15.2`.`